backend/mcp_backend/management/mcp_manager.py

```
@@ -0,0 +1,574 @@
+                "is_builtin": name
+                in ["time_server", "news_server"],  # Mark built-in servers
+            }
+            servers.append(server_info)
+
+        return servers
+
+    async def get_all_tools(self) -> List[Dict[str, Any]]:
+        """Get all tools from all connected MCP servers"""
+        all_tools = []
+
+        for server_name in self.server_configs.keys():
```

Ruff (0.12.2)

Use `key in dict` instead of `key in dict.keys()`

Remove `.keys()`

(SIM118)